app/clients/zendesk_sell.py

```diff
@@ -141,12 +151,55 @@ def upsert_lead(self, contact: ContactRequest) -> int:
 
         return resp.status_code
 
-    def upsert_contact(self, user: User) -> (Optional[int], bool):
+    def search_lead_id(self, user: User) -> Optional[str]:
+        resp, e = self._send_request(method='GET',
+                                     relative_url=f'/v2/leads?email={user.email_address}')
+        if e:
+            current_app.logger.warning('Failed to create zendesk sell contact')
+            return None
+
+        try:
+            # default to the first lead as we try to perform lead upsert
+            # There SHOULDN'T be any case where there is more than 1 entry
```

Comment:
WDYT about checking the length of items and make sure we fail if we have != 1 items

Reply:
Besides the automation/integration between Notify and Zendesk Sell, any Zendesk Sell user can still go in and synthetically duplicate leads. In this situation we do not want to fail as it's out of our control. The failure from my perspective doesn't add any value. We still want to convert the existing lead (even if there are numerous ones) to a contact to deduplicate the entry